### PR TITLE
[Merged by Bors] - added app tests for transactions

### PR DESCRIPTION
**Pull request by antonlerner**

### CODE REVIEW

File: cmd/node/app_test.go [hunk 1]
```diff
@@ -159,20 +160,197 @@ func activateGrpcServer(smApp *SpacemeshApp) {
...
+
+type TestScenario struct {
+	Setup        ScenarioSetup
+	Criteria     ScenarioTestCriteria
+	Dependencies []int
+}
+
+func txWithUnorderedNonceGenerator(dependancies []int) TestScenario {
+
+	acc1Signer, err := signing.NewEdSignerFromBuffer(util.FromHex(apiCfg.Account2Private))
```
Comment (noamnelke) [inline cmd/node/app_test.go:176]: Why is `acc1Signer` using `Account2Private`?

File: api/config/genesis.go [hunk 2]
```diff
@@ -55,12 +55,16 @@ func LoadGenesisConfig(path string) (*GenesisConfig, error) {
 const Account1Pub = "0x7be017a967db77fd10ac7c891b3d6d946dea7e3e14756e2f0f9e09b9663f0d9c"
 const Account1Private = "0x81c90dd832e18d1cf9758254327cb3135961af6688ac9c2a8c5d71f73acc5ce57be017a967db77fd10ac7c891b3d6d946dea7e3e14756e2f0f9e09b9663f0d9c"
 
+const Account2Pub = "0x22a31a84ab876f82fcafba86e77910b4419a4ee0f1d5483d7dd3b5b6b6922ee9"
+const Account2Private = "0x9d411020d46d3f4e1214f7b51052219737669f461ac9c9ac6ac49753926d0af222a31a84ab876f82fcafba86e77910b4419a4ee0f1d5483d7dd3b5b6b6922ee9"
```
Comment (noamnelke) [inline api/config/genesis.go:59]: Why is a second account needed?

**Comment by noamnelke** [comment]
Note that `appTest` is currently failing (also `go fmt ./...`)

File: cmd/node/app_test.go (hunk 1 above)
Comment (antonlerner) [reply cmd/node/app_test.go:176]: this is to seperate source accounts between tests

Comment (noamnelke) [reply cmd/node/app_test.go:176]: yeah, but doesn't it make more sense to call it `acc2Signer` if it uses Account 2's key?

**Comment by antonlerner** [comment]
@noamnelke I've fixed the app test to call getNonce before submiting transaction, please review

**Review by almogdepaz — APPROVED** [review]

**Comment by antonlerner** [comment]
bors +r

**Pull request closed without merging**